src/Uno.UWP/System/Launcher.cs

```diff
@@ -1,105 +1,70 @@
-﻿#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
-
-using System;
+﻿using System;
 using System.Threading.Tasks;
-using Uno.Extensions;
-using Uno.Logging;
 using Windows.Foundation;
+using Uno.Logging;
+using Uno.Extensions;
 using Microsoft.Extensions.Logging;
-using System.Linq;
 using Windows.UI.Core;
 
-#if __ANDROID__
-using Android.Content;
-using Android.Content.PM;
-#endif
-#if __IOS__
-using UIKit;
-using AppleUrl = global::Foundation.NSUrl;
-#endif
-
 namespace Windows.System
 {
 	public static partial class Launcher
 	{
-		public static async Task<bool> LaunchUriAsync(Uri uri)
+		private const string MicrosoftUriPrefix = "ms-";
+
+		private static bool IsSpecialUri(Uri uri) => uri.Scheme.StartsWith(MicrosoftUriPrefix, StringComparison.InvariantCultureIgnoreCase);
+
+
+		public static IAsyncOperation<bool> LaunchUriAsync(Uri uri)
 		{
-			return await CoreDispatcher.Main.RunWithResultAsync(
-				CoreDispatcherPriority.Normal,
-				async () =>
-				{
-					try
-					{
-#if __IOS__
-						return UIKit.UIApplication.SharedApplication.OpenUrl(
-							new AppleUrl(uri.OriginalString));
-#elif __ANDROID__
-						var androidUri = Android.Net.Uri.Parse(uri.OriginalString);
-						var intent = new Intent(Intent.ActionView, androidUri);
+#if __IOS__ || __ANDROID__ || __WASM__
+
+			if (uri == null)
+			{
+				// counterintuitively, UWP throws AccessViolationException when passed in Uri is null
+				throw new AccessViolationException("Attempted to read or write protected memory. This is often an indication that other memory is corrupt.");
```

Comment:
That seems more like a WinRT bug than anything else :)

Reply:
Especially when one method throws `AccessViolationException` and another throws `Exception`. Kind of confusing :-D .

Reply:
Fixed :-)